auth/src/main/java/com/firebase/ui/auth/AuthUI.java

```diff
@@ -262,6 +267,93 @@ public static int getDefaultTheme() {
         return R.style.FirebaseUI;
     }
 
+    /**
+     * Signs the user in without any UI if possible. If this operation fails, you can safely start a
+     * UI-based sign-in flow knowing it is required.
+     *
+     * @param context        requesting the user be signed in
+     * @param desiredConfigs to use for silent sign in. Only Google and email are currently
+     *                       supported, the rest will be ignored.
+     * @return a task which indicates whether or not the user was successfully signed in.
+     */
+    @NonNull
+    public Task<AuthResult> silentSignIn(@NonNull Context context,
```

Comment:
I am somewhat uncomfortable with all this code and chained async operations happening in one huge method block in `AuthUI`.  Should we consider making this happen inside a ViewModel in `KickoffActivity` or something so that it's rotation-immune?  Obviously then we couldn't directly return a `Task` but I think `onActivityResult` would be appropriate (and more consistent with our general mechanism anyway)

Reply:
Yeah, I don't like the number of chained tasks we have either, but I strongly disagree about the `onActivityResult` and `ViewModel` stuff. The point of this method is that it doesn't have any UI and should be done in the background (so it's up to the dev to deal with async callbacks as usual). As for rotation problems, they're up to the dev just like with any other `Task`. Also, I tried calling the method multiple times myself and it doesn't seem to matter besides being signed-in several times with the same credentials on the Firebase side (so the method is deterministic in that you'll just get the same result multiple times).

Do you see where I'm coming from? I literally call this method in my `Application#onCreate()` and I think other people will want to do something similar.